### PR TITLE
feature: paginated list results on API endpoints

### PR DESCRIPTION
**Pull request by lucasmenendez**
solving #99 

>Implement some pagination strategy over large list results on API endpoints. 
`GET /tokens`
`GET /strategies`
It is also needed for future endpoints, such as: 
`GET /tokens/{token_address}/holders`
`GET /censuses`

### CODE REVIEW

File: api/tokens.go [hunk 1]
```diff
@@ -51,21 +81,62 @@ func (capi *census3API) getTokens(msg *api.APIdata, ctx *httprouter.HTTPContext)
...
+		return ErrCantGetTokens.WithErr(err)
+	}
+	if nextCursor != nil {
+		tokensResponse.Pagination.NextCursor = common.BytesToAddress(nextCursor).String()
+	}
+	prevCursor, err := qtx.ListTokensPrevCursor(internalCtx,
+		queries.ListTokensPrevCursorParams{
+			PageCursor: currentCursor.Bytes(),
+			Limit:      pageSize,
+		})
```
Comment (mvdan) [inline api/tokens.go:114]: Hm, this is three queries for a single API request, which feels somewhat expensive.

Can the API client not use the first and last ID it received as cursors? For example, if the table has IDs:

```
1 2 3 4 5 6 7 8 9
```

right now you might return a response like:

```
items: [4 5 6]
next: 7
prev: 3
```

so that later you could allow the user to continue with `next` like `>= 7`.

However, you could return just `items: [4 5 6]`, and the user can give you the last ID, and you can continue the query with `> 6` instead.

basically, the pagination doesn't need to tell the user what the next and previous IDs are, outside of the query range. they can simply use "this is the smallest and largest ID I got to", and you can start the query from that point, excluding it.

Comment (lucasmenendez) [reply api/tokens.go:114]: This works for the next cursor, but the problem with this approach is with the previous cursor. For example, with `pageSize=3` and `cursor=3`, the results will be `[4, 5, 6]` (all ID's greater than the `cursor`). So the smallest ID of the results, the prev cursor, will return `[5, 6, 7]` instead of `[1, 2, 3]`.

It could be solve keeping two queries, one to get the results with the current cursor, using the last result as the next cursor, and another one to get the previous cursor.

But there is another problem with knowing whether you are on the last page or not, because if you are already on the last page and it has the same number of returns as the limit of rows, the result will contain a valid next cursor (the last ID), but if you try to get the page from that cursor, no rows will be returned.

**Comment by lucasmenendez** [comment]
@jordipainan if you are agree i will merge this today to deploy it to dev

**Review by jordipainan — APPROVED** [review]
LGTM, but reduce page size to 10. Strategies are not that big and dealing with 100 tokens at the same page is unnecessary. Thank you for the work :)
